### PR TITLE
Inject an activitysource

### PR DESCRIPTION
**Pull request by cartermp**
Fixes https://github.com/honeycombio/honeycomb-opentelemetry-dotnet/issues/212

So. I don't like the fact that .NET uses system diagnostics by default rather than the _actual_ OTel API nouns and verbs. I also really don't like dependency injection and believe that developers should just create singleton instances of these things.

But.

A lot of .NET developers like using dependency injection, and in some cases, they must use it (as it _can_ be the best solution to their problem). And a lot of .NET developers _want_ to use the system diagnostics APIs.

So, this enables them to do it with DI, which should allow them to do things like define a controller or API route that "magically" has an `ActivitySource` named after their service available to them. For example, our sample can use it like this:

```csharp
public WeatherForecastController(ILogger<WeatherForecastController> logger, ActivitySource source, Meter meter)
{
    _logger = logger;
    _source_ = source;
    _counter = meter.CreateCounter<int>("sheep");
}

//...

public async Task<IEnumerable<WeatherForecast>> Get()
{
    using (var activity = _source_.StartActivity("sleep"))
    {
        activity?.SetTag("delay_ms", 100);
        await Task.Delay(TimeSpan.FromMilliseconds(100));
    }

    _counter.Add(1);
    
    var rng = new Random();
    return Enumerable.Range(1, 5).Select(index => new WeatherForecast
    {
        Date = DateTime.Now.AddDays(index),
        TemperatureC = rng.Next(-20, 55),
        Summary = Summaries[rng.Next(Summaries.Length)]
    })
    .ToArray();
}
```

In other words, this meets more developers where they are, even if that place is not entirely aligned with my own idealogical beliefs 🙂.

I can also add a new sample to the repo that shows how to use this rather than the tracer. And I can also update docs.

### CODE REVIEW

**Comment by MikeGoldsmith** [comment]
@pkanal this may be related to what you're working on with adding support for `OTEL_SERVICE_NAME` as we need to synchronise the service named used both to create the activity source listener and registering the activity source with the DI.

**Comment by pkanal** [comment]
> @pkanal this may be related to what you're working on with adding support for `OTEL_SERVICE_NAME` as we need to synchronise the service named used both to create the activity source listener and registering the activity source with the DI.

We still rely on [`options.ServiceName`](https://github.com/honeycombio/honeycomb-opentelemetry-dotnet/pull/244/files#diff-2b876f482afe89e93d0188417e436fdfc03abd3ff8c0bffcbb652c02ee938b2dR68) here so if we don't have that set but the env var `OTEL_SERVICE_NAME` is set, `options` still doesn't have a way of knowing about it. 

Throughout this library it looks like we user `options` everywhere so I think if we want to reliably pull in env variables, we'll have to work it into overriding `options`. Let me know what you think of that approach!

**Comment by cartermp** [comment]
Some kind of environment variable detection would have to happen _before_ any call to `AddHoneycomb` is done when the app is run, and we'd have to fill out the service name based on that.

I wouldn't know the best place to try and detect that, but this is where I'd start: https://github.com/honeycombio/honeycomb-opentelemetry-dotnet/blob/main/src/Honeycomb.OpenTelemetry/HoneycombOptions.cs#L161

Or any place where a HoneycombOptions gets constructed (that is, create a parameter-less constructor that does this, and also do it in `FromArgs`). I think it would be best for us to do this early so that we have a strongly-typed value we can rely on everywhere else.

**Comment by MikeGoldsmith** [comment]
I agree a single consolidated object with all options on would be preferable, and it would know how to fall back between env var, app settings, programmatic options, etc.

**Comment by cartermp** [comment]
Hmmm, I wonder if this is a problem by registering two tracerproviders.

**Comment by MikeGoldsmith** [comment]
Yeah, after discussion on how hard this can be to debug. Maybe we add "not seeing telemetry trouble shooting" instead of registering an activity source for users.

**Review by MikeGoldsmith — CHANGES_REQUESTED** [review]
Maybe improve trouble shooting instead of being clever and registering activity source for users. Double registering an activity source is difficult to debug and will eat into customer event limits.

**Comment by cartermp** [comment]
After some discussion let's close this and see if there's a better overall approach to the problem space.
